h2integrate/converters/grid/grid_sell.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+import openmdao.api as om
+from attrs import field, define
+
+from h2integrate.core.utilities import BaseConfig, CostModelBaseConfig, merge_shared_inputs
+from h2integrate.core.model_baseclasses import CostModelBaseClass
+
+
+@define
+class GridSellPerformanceModelConfig(BaseConfig):
+    """Configuration for grid electricity selling performance model."""
+
+    interconnect_limit: float = field(default=1000000.0)  # kW
+
+
+class GridSellPerformanceModel(om.ExplicitComponent):
+    """
+    An OpenMDAO component that models selling electricity to the grid.
+
+    This component handles:
+    - Taking in excess electricity from plant generation
+    - Selling excess electricity to grid
+    - Calculating revenue based on sell price
+    - Enforcing interconnect limits
+    """
+
+    def initialize(self):
+        self.options.declare("driver_config", types=dict)
+        self.options.declare("plant_config", types=dict)
+        self.options.declare("tech_config", types=dict)
+
+    def setup(self):
+        self.config = GridSellPerformanceModelConfig.from_dict(
+            merge_shared_inputs(self.options["tech_config"]["model_inputs"], "performance")
+        )
+
+        n_timesteps = self.options["plant_config"]["plant"]["simulation"]["n_timesteps"]
+
+        # Inputs
+        self.add_input(
+            "electricity_in",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Excess electricity available to sell to grid",
+        )
+
+        self.add_input(
+            "interconnect_limit",
+            val=self.config.interconnect_limit,
+            units="kW",
+            desc="Maximum power that can be sold to grid",
+        )
+
+        # Outputs
+        self.add_output(
+            "electricity_consumed",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Electricity sold to grid (negative = selling)",
+        )
+
+    def compute(self, inputs, outputs):
+        electricity_in = inputs["electricity_in"]
+        interconnect_limit = inputs["interconnect_limit"]
+
+        # Amount to sell to grid (limited by interconnect and available electricity)
+        # Only sell positive electricity amounts
+        grid_sale = np.clip(electricity_in, 0, interconnect_limit)
+
+        outputs["electricity_consumed"] = -grid_sale
+
+
+@define
+class GridSellCostModelConfig(CostModelBaseConfig):
+    """Configuration for grid selling cost model."""
+
+    # Grid connections typically have minimal CapEx and OpEx
+    # Most revenue is operational (electricity sales) handled in performance model
+    start_up_cost: float = field(default=0.0)  # USD
```

Comment:
I think this should be connection cost or something similar that is more common (unless start_up_cost is common and outside my experience)